// pipedrive@23.4.3/dist/model/AddActivityResponseRelatedObjects.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _RelatedDealData = _interopRequireDefault(require("./RelatedDealData"));
var _RelatedOrganizationDataWithActiveFlag = _interopRequireDefault(require("./RelatedOrganizationDataWithActiveFlag"));
var _RelatedPersonDataWithActiveFlag = _interopRequireDefault(require("./RelatedPersonDataWithActiveFlag"));
var _RelatedUserData = _interopRequireDefault(require("./RelatedUserData"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The AddActivityResponseRelatedObjects model module.
 * @module model/AddActivityResponseRelatedObjects
 * @version 1.0.0
 */
var AddActivityResponseRelatedObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddActivityResponseRelatedObjects</code>.
   * @alias module:model/AddActivityResponseRelatedObjects
   */
  function AddActivityResponseRelatedObjects() {
    (0, _classCallCheck2["default"])(this, AddActivityResponseRelatedObjects);
    AddActivityResponseRelatedObjects.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(AddActivityResponseRelatedObjects, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AddActivityResponseRelatedObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddActivityResponseRelatedObjects} obj Optional instance to populate.
     * @return {module:model/AddActivityResponseRelatedObjects} The populated <code>AddActivityResponseRelatedObjects</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddActivityResponseRelatedObjects();
        if (data.hasOwnProperty('user')) {
          obj['user'] = _RelatedUserData["default"].constructFromObject(data['user']);
          delete data['user'];
        }
        if (data.hasOwnProperty('deal')) {
          obj['deal'] = _RelatedDealData["default"].constructFromObject(data['deal']);
          delete data['deal'];
        }
        if (data.hasOwnProperty('person')) {
          obj['person'] = _RelatedPersonDataWithActiveFlag["default"].constructFromObject(data['person']);
          delete data['person'];
        }
        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _RelatedOrganizationDataWithActiveFlag["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return AddActivityResponseRelatedObjects;
}();
/**
 * @member {module:model/RelatedUserData} user
 */
AddActivityResponseRelatedObjects.prototype['user'] = undefined;

/**
 * @member {module:model/RelatedDealData} deal
 */
AddActivityResponseRelatedObjects.prototype['deal'] = undefined;

/**
 * @member {module:model/RelatedPersonDataWithActiveFlag} person
 */
AddActivityResponseRelatedObjects.prototype['person'] = undefined;

/**
 * @member {module:model/RelatedOrganizationDataWithActiveFlag} organization
 */
AddActivityResponseRelatedObjects.prototype['organization'] = undefined;
var _default = AddActivityResponseRelatedObjects;
exports["default"] = _default;
