// pipedrive@23.4.3/dist/model/CallLogGoneResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The CallLogGoneResponse model module.
 * @module model/CallLogGoneResponse
 * @version 1.0.0
 */
var CallLogGoneResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CallLogGoneResponse</code>.
   * @alias module:model/CallLogGoneResponse
   */
  function CallLogGoneResponse() {
    (0, _classCallCheck2["default"])(this, CallLogGoneResponse);
    CallLogGoneResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(CallLogGoneResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>CallLogGoneResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CallLogGoneResponse} obj Optional instance to populate.
     * @return {module:model/CallLogGoneResponse} The populated <code>CallLogGoneResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CallLogGoneResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('error')) {
          obj['error'] = _ApiClient["default"].convertToType(data['error'], 'String');
          delete data['error'];
        }
        if (data.hasOwnProperty('error_info')) {
          obj['error_info'] = _ApiClient["default"].convertToType(data['error_info'], 'String');
          delete data['error_info'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], Object);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ApiClient["default"].convertToType(data['additional_data'], Object);
          delete data['additional_data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return CallLogGoneResponse;
}();
/**
 * @member {Boolean} success
 */
CallLogGoneResponse.prototype['success'] = undefined;

/**
 * The description of the error
 * @member {String} error
 */
CallLogGoneResponse.prototype['error'] = undefined;

/**
 * A message describing how to solve the problem
 * @member {String} error_info
 */
CallLogGoneResponse.prototype['error_info'] = undefined;

/**
 * @member {Object} data
 */
CallLogGoneResponse.prototype['data'] = undefined;

/**
 * @member {Object} additional_data
 */
CallLogGoneResponse.prototype['additional_data'] = undefined;
var _default = CallLogGoneResponse;
exports["default"] = _default;
