// pipedrive@23.4.3/dist/model/ChangelogResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AdditionalDataWithCursorPagination = _interopRequireDefault(require("./AdditionalDataWithCursorPagination"));
var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));
var _ChangelogResponseAllOf = _interopRequireDefault(require("./ChangelogResponseAllOf"));
var _ChangelogResponseAllOfData = _interopRequireDefault(require("./ChangelogResponseAllOfData"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ChangelogResponse model module.
 * @module model/ChangelogResponse
 * @version 1.0.0
 */
var ChangelogResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ChangelogResponse</code>.
   * @alias module:model/ChangelogResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ChangelogResponseAllOf
   */
  function ChangelogResponse() {
    (0, _classCallCheck2["default"])(this, ChangelogResponse);
    _BaseResponse["default"].initialize(this);
    _ChangelogResponseAllOf["default"].initialize(this);
    ChangelogResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(ChangelogResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ChangelogResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ChangelogResponse} obj Optional instance to populate.
     * @return {module:model/ChangelogResponse} The populated <code>ChangelogResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ChangelogResponse();
        _BaseResponse["default"].constructFromObject(data, obj);
        _ChangelogResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ChangelogResponseAllOfData["default"]]);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalDataWithCursorPagination["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return ChangelogResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */
ChangelogResponse.prototype['success'] = undefined;

/**
 * @member {Array.<module:model/ChangelogResponseAllOfData>} data
 */
ChangelogResponse.prototype['data'] = undefined;

/**
 * @member {module:model/AdditionalDataWithCursorPagination} additional_data
 */
ChangelogResponse.prototype['additional_data'] = undefined;

// Implement BaseResponse interface:
/**
 * If the response is successful or not
 * @member {Boolean} success
 */
_BaseResponse["default"].prototype['success'] = undefined;
// Implement ChangelogResponseAllOf interface:
/**
 * @member {Array.<module:model/ChangelogResponseAllOfData>} data
 */
_ChangelogResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalDataWithCursorPagination} additional_data
 */
_ChangelogResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = ChangelogResponse;
exports["default"] = _default;
