// pipedrive@23.4.3/dist/model/GetAddProductAttachmentDetails.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AddProductAttachmentDetails = _interopRequireDefault(require("./AddProductAttachmentDetails"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetAddProductAttachmentDetails model module.
 * @module model/GetAddProductAttachmentDetails
 * @version 1.0.0
 */
var GetAddProductAttachmentDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetAddProductAttachmentDetails</code>.
   * @alias module:model/GetAddProductAttachmentDetails
   */
  function GetAddProductAttachmentDetails() {
    (0, _classCallCheck2["default"])(this, GetAddProductAttachmentDetails);
    GetAddProductAttachmentDetails.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetAddProductAttachmentDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetAddProductAttachmentDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetAddProductAttachmentDetails} obj Optional instance to populate.
     * @return {module:model/GetAddProductAttachmentDetails} The populated <code>GetAddProductAttachmentDetails</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetAddProductAttachmentDetails();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], _AddProductAttachmentDetails["default"]);
          delete data['data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetAddProductAttachmentDetails;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */
GetAddProductAttachmentDetails.prototype['success'] = undefined;

/**
 * The added product object attached to the deal
 * @member {module:model/AddProductAttachmentDetails} data
 */
GetAddProductAttachmentDetails.prototype['data'] = undefined;
var _default = GetAddProductAttachmentDetails;
exports["default"] = _default;
