// pipedrive@23.4.3/dist/model/GetActivitiesResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));
var _GetActivitiesResponseRelatedObjects = _interopRequireDefault(require("./GetActivitiesResponseRelatedObjects"));
var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetActivitiesResponse model module.
 * @module model/GetActivitiesResponse
 * @version 1.0.0
 */
var GetActivitiesResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetActivitiesResponse</code>.
   * @alias module:model/GetActivitiesResponse
   */
  function GetActivitiesResponse() {
    (0, _classCallCheck2["default"])(this, GetActivitiesResponse);
    GetActivitiesResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetActivitiesResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetActivitiesResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetActivitiesResponse} obj Optional instance to populate.
     * @return {module:model/GetActivitiesResponse} The populated <code>GetActivitiesResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetActivitiesResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ActivityResponseObject["default"]]);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }
        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _GetActivitiesResponseRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetActivitiesResponse;
}();
/**
 * @member {Boolean} success
 */
GetActivitiesResponse.prototype['success'] = undefined;

/**
 * @member {Array.<module:model/ActivityResponseObject>} data
 */
GetActivitiesResponse.prototype['data'] = undefined;

/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */
GetActivitiesResponse.prototype['additional_data'] = undefined;

/**
 * @member {module:model/GetActivitiesResponseRelatedObjects} related_objects
 */
GetActivitiesResponse.prototype['related_objects'] = undefined;
var _default = GetActivitiesResponse;
exports["default"] = _default;
