// pipedrive@23.4.3/dist/model/GetActivityResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));
var _GetActivitiesResponseRelatedObjects = _interopRequireDefault(require("./GetActivitiesResponseRelatedObjects"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetActivityResponse model module.
 * @module model/GetActivityResponse
 * @version 1.0.0
 */
var GetActivityResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetActivityResponse</code>.
   * @alias module:model/GetActivityResponse
   */
  function GetActivityResponse() {
    (0, _classCallCheck2["default"])(this, GetActivityResponse);
    GetActivityResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetActivityResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetActivityResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetActivityResponse} obj Optional instance to populate.
     * @return {module:model/GetActivityResponse} The populated <code>GetActivityResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetActivityResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ActivityResponseObject["default"].constructFromObject(data['data']);
          delete data['data'];
        }
        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _GetActivitiesResponseRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetActivityResponse;
}();
/**
 * @member {Boolean} success
 */
GetActivityResponse.prototype['success'] = undefined;

/**
 * @member {module:model/ActivityResponseObject} data
 */
GetActivityResponse.prototype['data'] = undefined;

/**
 * @member {module:model/GetActivitiesResponseRelatedObjects} related_objects
 */
GetActivityResponse.prototype['related_objects'] = undefined;
var _default = GetActivityResponse;
exports["default"] = _default;
