// pipedrive@23.4.3/dist/model/GetFieldAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _InternalFieldType = _interopRequireDefault(require("./InternalFieldType"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetFieldAllOf model module.
 * @module model/GetFieldAllOf
 * @version 1.0.0
 */
var GetFieldAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetFieldAllOf</code>.
   * @alias module:model/GetFieldAllOf
   */
  function GetFieldAllOf() {
    (0, _classCallCheck2["default"])(this, GetFieldAllOf);
    GetFieldAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetFieldAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetFieldAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetFieldAllOf} obj Optional instance to populate.
     * @return {module:model/GetFieldAllOf} The populated <code>GetFieldAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetFieldAllOf();
        if (data.hasOwnProperty('field_type')) {
          obj['field_type'] = _InternalFieldType["default"].constructFromObject(data['field_type']);
          delete data['field_type'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetFieldAllOf;
}();
/**
 * @member {module:model/InternalFieldType} field_type
 */
GetFieldAllOf.prototype['field_type'] = undefined;
var _default = GetFieldAllOf;
exports["default"] = _default;
