// pipedrive@23.4.3/dist/model/GetProjectResponse200.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ProjectResponseObject = _interopRequireDefault(require("./ProjectResponseObject"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetProjectResponse200 model module.
 * @module model/GetProjectResponse200
 * @version 1.0.0
 */
var GetProjectResponse200 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetProjectResponse200</code>.
   * @alias module:model/GetProjectResponse200
   */
  function GetProjectResponse200() {
    (0, _classCallCheck2["default"])(this, GetProjectResponse200);
    GetProjectResponse200.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetProjectResponse200, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetProjectResponse200</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetProjectResponse200} obj Optional instance to populate.
     * @return {module:model/GetProjectResponse200} The populated <code>GetProjectResponse200</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetProjectResponse200();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ProjectResponseObject["default"].constructFromObject(data['data']);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ApiClient["default"].convertToType(data['additional_data'], Object);
          delete data['additional_data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetProjectResponse200;
}();
/**
 * @member {Boolean} success
 */
GetProjectResponse200.prototype['success'] = undefined;

/**
 * @member {module:model/ProjectResponseObject} data
 */
GetProjectResponse200.prototype['data'] = undefined;

/**
 * @member {Object} additional_data
 */
GetProjectResponse200.prototype['additional_data'] = undefined;
var _default = GetProjectResponse200;
exports["default"] = _default;
