// pipedrive@23.4.3/dist/model/GetProjectTemplatesResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AdditionalDataWithCursorPagination = _interopRequireDefault(require("./AdditionalDataWithCursorPagination"));
var _TemplateResponseObject = _interopRequireDefault(require("./TemplateResponseObject"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetProjectTemplatesResponse model module.
 * @module model/GetProjectTemplatesResponse
 * @version 1.0.0
 */
var GetProjectTemplatesResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetProjectTemplatesResponse</code>.
   * @alias module:model/GetProjectTemplatesResponse
   */
  function GetProjectTemplatesResponse() {
    (0, _classCallCheck2["default"])(this, GetProjectTemplatesResponse);
    GetProjectTemplatesResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetProjectTemplatesResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetProjectTemplatesResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetProjectTemplatesResponse} obj Optional instance to populate.
     * @return {module:model/GetProjectTemplatesResponse} The populated <code>GetProjectTemplatesResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetProjectTemplatesResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_TemplateResponseObject["default"]]);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalDataWithCursorPagination["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetProjectTemplatesResponse;
}();
/**
 * @member {Boolean} success
 */
GetProjectTemplatesResponse.prototype['success'] = undefined;

/**
 * @member {Array.<module:model/TemplateResponseObject>} data
 */
GetProjectTemplatesResponse.prototype['data'] = undefined;

/**
 * @member {module:model/AdditionalDataWithCursorPagination} additional_data
 */
GetProjectTemplatesResponse.prototype['additional_data'] = undefined;
var _default = GetProjectTemplatesResponse;
exports["default"] = _default;
