// pipedrive@23.4.3/dist/model/NewDealProductAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _BillingFrequency = _interopRequireDefault(require("./BillingFrequency"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The NewDealProductAllOf model module.
 * @module model/NewDealProductAllOf
 * @version 1.0.0
 */
var NewDealProductAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NewDealProductAllOf</code>.
   * @alias module:model/NewDealProductAllOf
   */
  function NewDealProductAllOf() {
    (0, _classCallCheck2["default"])(this, NewDealProductAllOf);
    NewDealProductAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(NewDealProductAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>NewDealProductAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NewDealProductAllOf} obj Optional instance to populate.
     * @return {module:model/NewDealProductAllOf} The populated <code>NewDealProductAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NewDealProductAllOf();
        if (data.hasOwnProperty('billing_frequency')) {
          obj['billing_frequency'] = _BillingFrequency["default"].constructFromObject(data['billing_frequency']);
          delete data['billing_frequency'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return NewDealProductAllOf;
}();
/**
 * @member {module:model/BillingFrequency} billing_frequency
 */
NewDealProductAllOf.prototype['billing_frequency'] = undefined;
var _default = NewDealProductAllOf;
exports["default"] = _default;
