// pipedrive@23.4.3/dist/model/NewDealProductAllOf1.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The NewDealProductAllOf1 model module.
 * @module model/NewDealProductAllOf1
 * @version 1.0.0
 */
var NewDealProductAllOf1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NewDealProductAllOf1</code>.
   * @alias module:model/NewDealProductAllOf1
   */
  function NewDealProductAllOf1() {
    (0, _classCallCheck2["default"])(this, NewDealProductAllOf1);
    NewDealProductAllOf1.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(NewDealProductAllOf1, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>NewDealProductAllOf1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NewDealProductAllOf1} obj Optional instance to populate.
     * @return {module:model/NewDealProductAllOf1} The populated <code>NewDealProductAllOf1</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NewDealProductAllOf1();
        if (data.hasOwnProperty('billing_frequency_cycles')) {
          obj['billing_frequency_cycles'] = _ApiClient["default"].convertToType(data['billing_frequency_cycles'], 'Number');
          delete data['billing_frequency_cycles'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return NewDealProductAllOf1;
}();
/**
 * Only available in Advanced and above plans  The number of times the billing frequency repeats for a product in a deal  When `billing_frequency` is set to `one-time`, this field must be `null`  For all the other values of `billing_frequency`, `null` represents a product billed indefinitely  Must be a positive integer less or equal to 312 
 * @member {Number} billing_frequency_cycles
 */
NewDealProductAllOf1.prototype['billing_frequency_cycles'] = undefined;
var _default = NewDealProductAllOf1;
exports["default"] = _default;
