// pipedrive@23.4.3/dist/model/ParticipantsChangelogItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DealParticipantsChangelog = _interopRequireDefault(require("./DealParticipantsChangelog"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ParticipantsChangelogItem model module.
 * @module model/ParticipantsChangelogItem
 * @version 1.0.0
 */
var ParticipantsChangelogItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ParticipantsChangelogItem</code>.
   * @alias module:model/ParticipantsChangelogItem
   * @implements module:model/DealParticipantsChangelog
   */
  function ParticipantsChangelogItem() {
    (0, _classCallCheck2["default"])(this, ParticipantsChangelogItem);
    _DealParticipantsChangelog["default"].initialize(this);
    ParticipantsChangelogItem.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(ParticipantsChangelogItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ParticipantsChangelogItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ParticipantsChangelogItem} obj Optional instance to populate.
     * @return {module:model/ParticipantsChangelogItem} The populated <code>ParticipantsChangelogItem</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ParticipantsChangelogItem();
        _DealParticipantsChangelog["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('actor_user_id')) {
          obj['actor_user_id'] = _ApiClient["default"].convertToType(data['actor_user_id'], 'Number');
          delete data['actor_user_id'];
        }
        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }
        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
          delete data['action'];
        }
        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'String');
          delete data['time'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return ParticipantsChangelogItem;
}();
/**
 * The ID of the user
 * @member {Number} actor_user_id
 */
ParticipantsChangelogItem.prototype['actor_user_id'] = undefined;

/**
 * The ID of the person
 * @member {Number} person_id
 */
ParticipantsChangelogItem.prototype['person_id'] = undefined;

/**
 * Deal participant action type
 * @member {String} action
 */
ParticipantsChangelogItem.prototype['action'] = undefined;

/**
 * The deal participant action log time
 * @member {String} time
 */
ParticipantsChangelogItem.prototype['time'] = undefined;

// Implement DealParticipantsChangelog interface:
/**
 * The ID of the user
 * @member {Number} actor_user_id
 */
_DealParticipantsChangelog["default"].prototype['actor_user_id'] = undefined;
/**
 * The ID of the person
 * @member {Number} person_id
 */
_DealParticipantsChangelog["default"].prototype['person_id'] = undefined;
/**
 * Deal participant action type
 * @member {String} action
 */
_DealParticipantsChangelog["default"].prototype['action'] = undefined;
/**
 * The deal participant action log time
 * @member {String} time
 */
_DealParticipantsChangelog["default"].prototype['time'] = undefined;
var _default = ParticipantsChangelogItem;
exports["default"] = _default;
