// pipedrive@23.4.3/dist/model/PostDealParticipantsRelatedObjects.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _RelatedPersonDataWithActiveFlag = _interopRequireDefault(require("./RelatedPersonDataWithActiveFlag"));
var _RelatedUserData = _interopRequireDefault(require("./RelatedUserData"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The PostDealParticipantsRelatedObjects model module.
 * @module model/PostDealParticipantsRelatedObjects
 * @version 1.0.0
 */
var PostDealParticipantsRelatedObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PostDealParticipantsRelatedObjects</code>.
   * @alias module:model/PostDealParticipantsRelatedObjects
   */
  function PostDealParticipantsRelatedObjects() {
    (0, _classCallCheck2["default"])(this, PostDealParticipantsRelatedObjects);
    PostDealParticipantsRelatedObjects.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(PostDealParticipantsRelatedObjects, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PostDealParticipantsRelatedObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostDealParticipantsRelatedObjects} obj Optional instance to populate.
     * @return {module:model/PostDealParticipantsRelatedObjects} The populated <code>PostDealParticipantsRelatedObjects</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PostDealParticipantsRelatedObjects();
        if (data.hasOwnProperty('user')) {
          obj['user'] = _RelatedUserData["default"].constructFromObject(data['user']);
          delete data['user'];
        }
        if (data.hasOwnProperty('person')) {
          obj['person'] = _RelatedPersonDataWithActiveFlag["default"].constructFromObject(data['person']);
          delete data['person'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return PostDealParticipantsRelatedObjects;
}();
/**
 * @member {module:model/RelatedUserData} user
 */
PostDealParticipantsRelatedObjects.prototype['user'] = undefined;

/**
 * @member {module:model/RelatedPersonDataWithActiveFlag} person
 */
PostDealParticipantsRelatedObjects.prototype['person'] = undefined;
var _default = PostDealParticipantsRelatedObjects;
exports["default"] = _default;
