// pipedrive@23.4.3/dist/model/ProjectGroupsObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ProjectGroupsObject model module.
 * @module model/ProjectGroupsObject
 * @version 1.0.0
 */
var ProjectGroupsObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectGroupsObject</code>.
   * @alias module:model/ProjectGroupsObject
   */
  function ProjectGroupsObject() {
    (0, _classCallCheck2["default"])(this, ProjectGroupsObject);
    ProjectGroupsObject.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(ProjectGroupsObject, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ProjectGroupsObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectGroupsObject} obj Optional instance to populate.
     * @return {module:model/ProjectGroupsObject} The populated <code>ProjectGroupsObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectGroupsObject();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }
        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return ProjectGroupsObject;
}();
/**
 * ID of the group
 * @member {Number} id
 */
ProjectGroupsObject.prototype['id'] = undefined;

/**
 * Name of the group
 * @member {String} name
 */
ProjectGroupsObject.prototype['name'] = undefined;

/**
 * Order number of the group
 * @member {Number} order_nr
 */
ProjectGroupsObject.prototype['order_nr'] = undefined;
var _default = ProjectGroupsObject;
exports["default"] = _default;
