// pipedrive@23.4.3/dist/model/ProjectPostObjectAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ProjectPostObjectAllOf model module.
 * @module model/ProjectPostObjectAllOf
 * @version 1.0.0
 */
var ProjectPostObjectAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectPostObjectAllOf</code>.
   * @alias module:model/ProjectPostObjectAllOf
   */
  function ProjectPostObjectAllOf() {
    (0, _classCallCheck2["default"])(this, ProjectPostObjectAllOf);
    ProjectPostObjectAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(ProjectPostObjectAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ProjectPostObjectAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectPostObjectAllOf} obj Optional instance to populate.
     * @return {module:model/ProjectPostObjectAllOf} The populated <code>ProjectPostObjectAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectPostObjectAllOf();
        if (data.hasOwnProperty('template_id')) {
          obj['template_id'] = _ApiClient["default"].convertToType(data['template_id'], 'Number');
          delete data['template_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return ProjectPostObjectAllOf;
}();
/**
 * The ID of the template the project will be based on
 * @member {Number} template_id
 */
ProjectPostObjectAllOf.prototype['template_id'] = undefined;
var _default = ProjectPostObjectAllOf;
exports["default"] = _default;
