// pipedrive@23.4.3/dist/model/ProjectMandatoryObjectFragment.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ProjectMandatoryObjectFragment model module.
 * @module model/ProjectMandatoryObjectFragment
 * @version 1.0.0
 */
var ProjectMandatoryObjectFragment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectMandatoryObjectFragment</code>.
   * @alias module:model/ProjectMandatoryObjectFragment
   */
  function ProjectMandatoryObjectFragment() {
    (0, _classCallCheck2["default"])(this, ProjectMandatoryObjectFragment);
    ProjectMandatoryObjectFragment.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(ProjectMandatoryObjectFragment, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ProjectMandatoryObjectFragment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectMandatoryObjectFragment} obj Optional instance to populate.
     * @return {module:model/ProjectMandatoryObjectFragment} The populated <code>ProjectMandatoryObjectFragment</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectMandatoryObjectFragment();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }
        if (data.hasOwnProperty('board_id')) {
          obj['board_id'] = _ApiClient["default"].convertToType(data['board_id'], 'Number');
          delete data['board_id'];
        }
        if (data.hasOwnProperty('phase_id')) {
          obj['phase_id'] = _ApiClient["default"].convertToType(data['phase_id'], 'Number');
          delete data['phase_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return ProjectMandatoryObjectFragment;
}();
/**
 * The title of the project
 * @member {String} title
 */
ProjectMandatoryObjectFragment.prototype['title'] = undefined;

/**
 * The ID of the board this project is associated with
 * @member {Number} board_id
 */
ProjectMandatoryObjectFragment.prototype['board_id'] = undefined;

/**
 * The ID of the phase this project is associated with
 * @member {Number} phase_id
 */
ProjectMandatoryObjectFragment.prototype['phase_id'] = undefined;
var _default = ProjectMandatoryObjectFragment;
exports["default"] = _default;
