// pipedrive@23.4.3/dist/model/ProjectPutObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ProjectMandatoryObjectFragment = _interopRequireDefault(require("./ProjectMandatoryObjectFragment"));
var _ProjectObjectFragment = _interopRequireDefault(require("./ProjectObjectFragment"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ProjectPutObject model module.
 * @module model/ProjectPutObject
 * @version 1.0.0
 */
var ProjectPutObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectPutObject</code>.
   * @alias module:model/ProjectPutObject
   * @implements module:model/ProjectMandatoryObjectFragment
   * @implements module:model/ProjectObjectFragment
   */
  function ProjectPutObject() {
    (0, _classCallCheck2["default"])(this, ProjectPutObject);
    _ProjectMandatoryObjectFragment["default"].initialize(this);
    _ProjectObjectFragment["default"].initialize(this);
    ProjectPutObject.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(ProjectPutObject, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ProjectPutObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectPutObject} obj Optional instance to populate.
     * @return {module:model/ProjectPutObject} The populated <code>ProjectPutObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectPutObject();
        _ProjectMandatoryObjectFragment["default"].constructFromObject(data, obj);
        _ProjectObjectFragment["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }
        if (data.hasOwnProperty('board_id')) {
          obj['board_id'] = _ApiClient["default"].convertToType(data['board_id'], 'Number');
          delete data['board_id'];
        }
        if (data.hasOwnProperty('phase_id')) {
          obj['phase_id'] = _ApiClient["default"].convertToType(data['phase_id'], 'Number');
          delete data['phase_id'];
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
          delete data['status'];
        }
        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }
        if (data.hasOwnProperty('start_date')) {
          obj['start_date'] = _ApiClient["default"].convertToType(data['start_date'], 'Date');
          delete data['start_date'];
        }
        if (data.hasOwnProperty('end_date')) {
          obj['end_date'] = _ApiClient["default"].convertToType(data['end_date'], 'Date');
          delete data['end_date'];
        }
        if (data.hasOwnProperty('deal_ids')) {
          obj['deal_ids'] = _ApiClient["default"].convertToType(data['deal_ids'], ['Number']);
          delete data['deal_ids'];
        }
        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }
        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }
        if (data.hasOwnProperty('labels')) {
          obj['labels'] = _ApiClient["default"].convertToType(data['labels'], ['Number']);
          delete data['labels'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return ProjectPutObject;
}();
/**
 * The title of the project
 * @member {String} title
 */
ProjectPutObject.prototype['title'] = undefined;

/**
 * The ID of the board this project is associated with
 * @member {Number} board_id
 */
ProjectPutObject.prototype['board_id'] = undefined;

/**
 * The ID of the phase this project is associated with
 * @member {Number} phase_id
 */
ProjectPutObject.prototype['phase_id'] = undefined;

/**
 * The description of the project
 * @member {String} description
 */
ProjectPutObject.prototype['description'] = undefined;

/**
 * The status of the project
 * @member {String} status
 */
ProjectPutObject.prototype['status'] = undefined;

/**
 * The ID of a project owner
 * @member {Number} owner_id
 */
ProjectPutObject.prototype['owner_id'] = undefined;

/**
 * The start date of the project. Format: YYYY-MM-DD.
 * @member {Date} start_date
 */
ProjectPutObject.prototype['start_date'] = undefined;

/**
 * The end date of the project. Format: YYYY-MM-DD.
 * @member {Date} end_date
 */
ProjectPutObject.prototype['end_date'] = undefined;

/**
 * An array of IDs of the deals this project is associated with
 * @member {Array.<Number>} deal_ids
 */
ProjectPutObject.prototype['deal_ids'] = undefined;

/**
 * The ID of the organization this project is associated with
 * @member {Number} org_id
 */
ProjectPutObject.prototype['org_id'] = undefined;

/**
 * The ID of the person this project is associated with
 * @member {Number} person_id
 */
ProjectPutObject.prototype['person_id'] = undefined;

/**
 * An array of IDs of the labels this project has
 * @member {Array.<Number>} labels
 */
ProjectPutObject.prototype['labels'] = undefined;

// Implement ProjectMandatoryObjectFragment interface:
/**
 * The title of the project
 * @member {String} title
 */
_ProjectMandatoryObjectFragment["default"].prototype['title'] = undefined;
/**
 * The ID of the board this project is associated with
 * @member {Number} board_id
 */
_ProjectMandatoryObjectFragment["default"].prototype['board_id'] = undefined;
/**
 * The ID of the phase this project is associated with
 * @member {Number} phase_id
 */
_ProjectMandatoryObjectFragment["default"].prototype['phase_id'] = undefined;
// Implement ProjectObjectFragment interface:
/**
 * The description of the project
 * @member {String} description
 */
_ProjectObjectFragment["default"].prototype['description'] = undefined;
/**
 * The status of the project
 * @member {String} status
 */
_ProjectObjectFragment["default"].prototype['status'] = undefined;
/**
 * The ID of a project owner
 * @member {Number} owner_id
 */
_ProjectObjectFragment["default"].prototype['owner_id'] = undefined;
/**
 * The start date of the project. Format: YYYY-MM-DD.
 * @member {Date} start_date
 */
_ProjectObjectFragment["default"].prototype['start_date'] = undefined;
/**
 * The end date of the project. Format: YYYY-MM-DD.
 * @member {Date} end_date
 */
_ProjectObjectFragment["default"].prototype['end_date'] = undefined;
/**
 * An array of IDs of the deals this project is associated with
 * @member {Array.<Number>} deal_ids
 */
_ProjectObjectFragment["default"].prototype['deal_ids'] = undefined;
/**
 * The ID of the organization this project is associated with
 * @member {Number} org_id
 */
_ProjectObjectFragment["default"].prototype['org_id'] = undefined;
/**
 * The ID of the person this project is associated with
 * @member {Number} person_id
 */
_ProjectObjectFragment["default"].prototype['person_id'] = undefined;
/**
 * An array of IDs of the labels this project has
 * @member {Array.<Number>} labels
 */
_ProjectObjectFragment["default"].prototype['labels'] = undefined;
var _default = ProjectPutObject;
exports["default"] = _default;
