// pipedrive@23.4.3/dist/model/RequiredPostTaskParameters.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The RequiredPostTaskParameters model module.
 * @module model/RequiredPostTaskParameters
 * @version 1.0.0
 */
var RequiredPostTaskParameters = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RequiredPostTaskParameters</code>.
   * @alias module:model/RequiredPostTaskParameters
   * @param title {String} The title of the task
   * @param projectId {Number} The ID of a project
   */
  function RequiredPostTaskParameters(title, projectId) {
    (0, _classCallCheck2["default"])(this, RequiredPostTaskParameters);
    RequiredPostTaskParameters.initialize(this, title, projectId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(RequiredPostTaskParameters, null, [{
    key: "initialize",
    value: function initialize(obj, title, projectId) {
      obj['title'] = title;
      obj['project_id'] = projectId;
    }

    /**
     * Constructs a <code>RequiredPostTaskParameters</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RequiredPostTaskParameters} obj Optional instance to populate.
     * @return {module:model/RequiredPostTaskParameters} The populated <code>RequiredPostTaskParameters</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RequiredPostTaskParameters();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
          delete data['project_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return RequiredPostTaskParameters;
}();
/**
 * The title of the task
 * @member {String} title
 */
RequiredPostTaskParameters.prototype['title'] = undefined;

/**
 * The ID of a project
 * @member {Number} project_id
 */
RequiredPostTaskParameters.prototype['project_id'] = undefined;
var _default = RequiredPostTaskParameters;
exports["default"] = _default;
