// pipedrive@23.4.3/dist/model/TaskResponseObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _FullTaskObject = _interopRequireDefault(require("./FullTaskObject"));
var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));
var _TaskId = _interopRequireDefault(require("./TaskId"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The TaskResponseObject model module.
 * @module model/TaskResponseObject
 * @version 1.0.0
 */
var TaskResponseObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TaskResponseObject</code>.
   * @alias module:model/TaskResponseObject
   * @implements module:model/TaskId
   * @implements module:model/FullTaskObject
   */
  function TaskResponseObject() {
    (0, _classCallCheck2["default"])(this, TaskResponseObject);
    _TaskId["default"].initialize(this);
    _FullTaskObject["default"].initialize(this);
    TaskResponseObject.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(TaskResponseObject, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TaskResponseObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TaskResponseObject} obj Optional instance to populate.
     * @return {module:model/TaskResponseObject} The populated <code>TaskResponseObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TaskResponseObject();
        _TaskId["default"].constructFromObject(data, obj);
        _FullTaskObject["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
          delete data['project_id'];
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }
        if (data.hasOwnProperty('parent_task_id')) {
          obj['parent_task_id'] = _ApiClient["default"].convertToType(data['parent_task_id'], 'Number');
          delete data['parent_task_id'];
        }
        if (data.hasOwnProperty('assignee_id')) {
          obj['assignee_id'] = _ApiClient["default"].convertToType(data['assignee_id'], 'Number');
          delete data['assignee_id'];
        }
        if (data.hasOwnProperty('done')) {
          obj['done'] = _ApiClient["default"].convertToType(data['done'], _NumberBoolean["default"]);
          delete data['done'];
        }
        if (data.hasOwnProperty('due_date')) {
          obj['due_date'] = _ApiClient["default"].convertToType(data['due_date'], 'Date');
          delete data['due_date'];
        }
        if (data.hasOwnProperty('creator_id')) {
          obj['creator_id'] = _ApiClient["default"].convertToType(data['creator_id'], 'Number');
          delete data['creator_id'];
        }
        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }
        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }
        if (data.hasOwnProperty('marked_as_done_time')) {
          obj['marked_as_done_time'] = _ApiClient["default"].convertToType(data['marked_as_done_time'], 'String');
          delete data['marked_as_done_time'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return TaskResponseObject;
}();
/**
 * The ID of the task, generated when the task was created
 * @member {Number} id
 */
TaskResponseObject.prototype['id'] = undefined;

/**
 * The title of the task
 * @member {String} title
 */
TaskResponseObject.prototype['title'] = undefined;

/**
 * The ID of the project this task is associated with
 * @member {Number} project_id
 */
TaskResponseObject.prototype['project_id'] = undefined;

/**
 * The description of the task
 * @member {String} description
 */
TaskResponseObject.prototype['description'] = undefined;

/**
 * The ID of a parent task. Can not be ID of a task which is already a subtask.
 * @member {Number} parent_task_id
 */
TaskResponseObject.prototype['parent_task_id'] = undefined;

/**
 * The ID of the user who will be the assignee of the task
 * @member {Number} assignee_id
 */
TaskResponseObject.prototype['assignee_id'] = undefined;

/**
 * Whether the task is done or not. 0 = Not done, 1 = Done.
 * @member {module:model/NumberBoolean} done
 */
TaskResponseObject.prototype['done'] = undefined;

/**
 * The due date of the task. Format: YYYY-MM-DD.
 * @member {Date} due_date
 */
TaskResponseObject.prototype['due_date'] = undefined;

/**
 * The creator of a task
 * @member {Number} creator_id
 */
TaskResponseObject.prototype['creator_id'] = undefined;

/**
 * The creation date and time of the task in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} add_time
 */
TaskResponseObject.prototype['add_time'] = undefined;

/**
 * The update date and time of the task in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} update_time
 */
TaskResponseObject.prototype['update_time'] = undefined;

/**
 * The marked as done date and time of the task in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} marked_as_done_time
 */
TaskResponseObject.prototype['marked_as_done_time'] = undefined;

// Implement TaskId interface:
/**
 * The ID of the task, generated when the task was created
 * @member {Number} id
 */
_TaskId["default"].prototype['id'] = undefined;
// Implement FullTaskObject interface:
/**
 * The title of the task
 * @member {String} title
 */
_FullTaskObject["default"].prototype['title'] = undefined;
/**
 * The ID of the project this task is associated with
 * @member {Number} project_id
 */
_FullTaskObject["default"].prototype['project_id'] = undefined;
/**
 * The description of the task
 * @member {String} description
 */
_FullTaskObject["default"].prototype['description'] = undefined;
/**
 * The ID of a parent task. Can not be ID of a task which is already a subtask.
 * @member {Number} parent_task_id
 */
_FullTaskObject["default"].prototype['parent_task_id'] = undefined;
/**
 * The ID of the user who will be the assignee of the task
 * @member {Number} assignee_id
 */
_FullTaskObject["default"].prototype['assignee_id'] = undefined;
/**
 * Whether the task is done or not. 0 = Not done, 1 = Done.
 * @member {module:model/NumberBoolean} done
 */
_FullTaskObject["default"].prototype['done'] = undefined;
/**
 * The due date of the task. Format: YYYY-MM-DD.
 * @member {Date} due_date
 */
_FullTaskObject["default"].prototype['due_date'] = undefined;
/**
 * The creator of a task
 * @member {Number} creator_id
 */
_FullTaskObject["default"].prototype['creator_id'] = undefined;
/**
 * The creation date and time of the task in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} add_time
 */
_FullTaskObject["default"].prototype['add_time'] = undefined;
/**
 * The update date and time of the task in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} update_time
 */
_FullTaskObject["default"].prototype['update_time'] = undefined;
/**
 * The marked as done date and time of the task in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} marked_as_done_time
 */
_FullTaskObject["default"].prototype['marked_as_done_time'] = undefined;
var _default = TaskResponseObject;
exports["default"] = _default;
