// pipedrive@23.4.3/dist/model/TaskPostObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));
var _RequiredPostTaskParameters = _interopRequireDefault(require("./RequiredPostTaskParameters"));
var _TaskObjectFragment = _interopRequireDefault(require("./TaskObjectFragment"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The TaskPostObject model module.
 * @module model/TaskPostObject
 * @version 1.0.0
 */
var TaskPostObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TaskPostObject</code>.
   * @alias module:model/TaskPostObject
   * @implements module:model/RequiredPostTaskParameters
   * @implements module:model/TaskObjectFragment
   * @param title {String} The title of the task
   * @param projectId {Number} The ID of a project
   */
  function TaskPostObject(title, projectId) {
    (0, _classCallCheck2["default"])(this, TaskPostObject);
    _RequiredPostTaskParameters["default"].initialize(this, title, projectId);
    _TaskObjectFragment["default"].initialize(this);
    TaskPostObject.initialize(this, title, projectId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(TaskPostObject, null, [{
    key: "initialize",
    value: function initialize(obj, title, projectId) {
      obj['title'] = title;
      obj['project_id'] = projectId;
    }

    /**
     * Constructs a <code>TaskPostObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TaskPostObject} obj Optional instance to populate.
     * @return {module:model/TaskPostObject} The populated <code>TaskPostObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TaskPostObject();
        _RequiredPostTaskParameters["default"].constructFromObject(data, obj);
        _TaskObjectFragment["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
          delete data['project_id'];
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }
        if (data.hasOwnProperty('parent_task_id')) {
          obj['parent_task_id'] = _ApiClient["default"].convertToType(data['parent_task_id'], 'Number');
          delete data['parent_task_id'];
        }
        if (data.hasOwnProperty('assignee_id')) {
          obj['assignee_id'] = _ApiClient["default"].convertToType(data['assignee_id'], 'Number');
          delete data['assignee_id'];
        }
        if (data.hasOwnProperty('done')) {
          obj['done'] = _ApiClient["default"].convertToType(data['done'], _NumberBoolean["default"]);
          delete data['done'];
        }
        if (data.hasOwnProperty('due_date')) {
          obj['due_date'] = _ApiClient["default"].convertToType(data['due_date'], 'Date');
          delete data['due_date'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return TaskPostObject;
}();
/**
 * The title of the task
 * @member {String} title
 */
TaskPostObject.prototype['title'] = undefined;

/**
 * The ID of a project
 * @member {Number} project_id
 */
TaskPostObject.prototype['project_id'] = undefined;

/**
 * The description of the task
 * @member {String} description
 */
TaskPostObject.prototype['description'] = undefined;

/**
 * The ID of a parent task. Can not be ID of a task which is already a subtask.
 * @member {Number} parent_task_id
 */
TaskPostObject.prototype['parent_task_id'] = undefined;

/**
 * The ID of the user who will be the assignee of the task
 * @member {Number} assignee_id
 */
TaskPostObject.prototype['assignee_id'] = undefined;

/**
 * Whether the task is done or not. 0 = Not done, 1 = Done.
 * @member {module:model/NumberBoolean} done
 */
TaskPostObject.prototype['done'] = undefined;

/**
 * The due date of the task. Format: YYYY-MM-DD.
 * @member {Date} due_date
 */
TaskPostObject.prototype['due_date'] = undefined;

// Implement RequiredPostTaskParameters interface:
/**
 * The title of the task
 * @member {String} title
 */
_RequiredPostTaskParameters["default"].prototype['title'] = undefined;
/**
 * The ID of a project
 * @member {Number} project_id
 */
_RequiredPostTaskParameters["default"].prototype['project_id'] = undefined;
// Implement TaskObjectFragment interface:
/**
 * The description of the task
 * @member {String} description
 */
_TaskObjectFragment["default"].prototype['description'] = undefined;
/**
 * The ID of a parent task. Can not be ID of a task which is already a subtask.
 * @member {Number} parent_task_id
 */
_TaskObjectFragment["default"].prototype['parent_task_id'] = undefined;
/**
 * The ID of the user who will be the assignee of the task
 * @member {Number} assignee_id
 */
_TaskObjectFragment["default"].prototype['assignee_id'] = undefined;
/**
 * Whether the task is done or not. 0 = Not done, 1 = Done.
 * @member {module:model/NumberBoolean} done
 */
_TaskObjectFragment["default"].prototype['done'] = undefined;
/**
 * The due date of the task. Format: YYYY-MM-DD.
 * @member {Date} due_date
 */
_TaskObjectFragment["default"].prototype['due_date'] = undefined;
var _default = TaskPostObject;
exports["default"] = _default;
