// pipedrive@23.4.3/dist/model/TaskMandatoryObjectFragment.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The TaskMandatoryObjectFragment model module.
 * @module model/TaskMandatoryObjectFragment
 * @version 1.0.0
 */
var TaskMandatoryObjectFragment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TaskMandatoryObjectFragment</code>.
   * @alias module:model/TaskMandatoryObjectFragment
   */
  function TaskMandatoryObjectFragment() {
    (0, _classCallCheck2["default"])(this, TaskMandatoryObjectFragment);
    TaskMandatoryObjectFragment.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(TaskMandatoryObjectFragment, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TaskMandatoryObjectFragment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TaskMandatoryObjectFragment} obj Optional instance to populate.
     * @return {module:model/TaskMandatoryObjectFragment} The populated <code>TaskMandatoryObjectFragment</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TaskMandatoryObjectFragment();
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
          delete data['project_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return TaskMandatoryObjectFragment;
}();
/**
 * The title of the task
 * @member {String} title
 */
TaskMandatoryObjectFragment.prototype['title'] = undefined;

/**
 * The ID of the project this task is associated with
 * @member {Number} project_id
 */
TaskMandatoryObjectFragment.prototype['project_id'] = undefined;
var _default = TaskMandatoryObjectFragment;
exports["default"] = _default;
