// pipedrive@23.4.3/dist/model/UpdateActivityResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));
var _AddActivityResponseRelatedObjects = _interopRequireDefault(require("./AddActivityResponseRelatedObjects"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UpdateActivityResponse model module.
 * @module model/UpdateActivityResponse
 * @version 1.0.0
 */
var UpdateActivityResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateActivityResponse</code>.
   * @alias module:model/UpdateActivityResponse
   */
  function UpdateActivityResponse() {
    (0, _classCallCheck2["default"])(this, UpdateActivityResponse);
    UpdateActivityResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UpdateActivityResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UpdateActivityResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateActivityResponse} obj Optional instance to populate.
     * @return {module:model/UpdateActivityResponse} The populated <code>UpdateActivityResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateActivityResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ActivityResponseObject["default"].constructFromObject(data['data']);
          delete data['data'];
        }
        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AddActivityResponseRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UpdateActivityResponse;
}();
/**
 * @member {Boolean} success
 */
UpdateActivityResponse.prototype['success'] = undefined;

/**
 * @member {module:model/ActivityResponseObject} data
 */
UpdateActivityResponse.prototype['data'] = undefined;

/**
 * @member {module:model/AddActivityResponseRelatedObjects} related_objects
 */
UpdateActivityResponse.prototype['related_objects'] = undefined;
var _default = UpdateActivityResponse;
exports["default"] = _default;
