// pipedrive@23.4.3/dist/model/UpsertLeadLabelResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _LeadLabelResponse = _interopRequireDefault(require("./LeadLabelResponse"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UpsertLeadLabelResponse model module.
 * @module model/UpsertLeadLabelResponse
 * @version 1.0.0
 */
var UpsertLeadLabelResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpsertLeadLabelResponse</code>.
   * @alias module:model/UpsertLeadLabelResponse
   */
  function UpsertLeadLabelResponse() {
    (0, _classCallCheck2["default"])(this, UpsertLeadLabelResponse);
    UpsertLeadLabelResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UpsertLeadLabelResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UpsertLeadLabelResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpsertLeadLabelResponse} obj Optional instance to populate.
     * @return {module:model/UpsertLeadLabelResponse} The populated <code>UpsertLeadLabelResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpsertLeadLabelResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _LeadLabelResponse["default"].constructFromObject(data['data']);
          delete data['data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UpsertLeadLabelResponse;
}();
/**
 * @member {Boolean} success
 */
UpsertLeadLabelResponse.prototype['success'] = undefined;

/**
 * @member {module:model/LeadLabelResponse} data
 */
UpsertLeadLabelResponse.prototype['data'] = undefined;
var _default = UpsertLeadLabelResponse;
exports["default"] = _default;
