// pipedrive@23.4.3/dist/model/UserProviderLinkCreateRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UserProviderLinkCreateRequest model module.
 * @module model/UserProviderLinkCreateRequest
 * @version 1.0.0
 */
var UserProviderLinkCreateRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserProviderLinkCreateRequest</code>.
   * @alias module:model/UserProviderLinkCreateRequest
   * @param userProviderId {String} Unique identifier linking a user to the installed integration. Generated by the integration.
   * @param userId {Number} Pipedrive user ID
   * @param companyId {Number} Pipedrive company ID
   * @param marketplaceClientId {String} Pipedrive Marketplace client ID of the installed integration
   */
  function UserProviderLinkCreateRequest(userProviderId, userId, companyId, marketplaceClientId) {
    (0, _classCallCheck2["default"])(this, UserProviderLinkCreateRequest);
    UserProviderLinkCreateRequest.initialize(this, userProviderId, userId, companyId, marketplaceClientId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UserProviderLinkCreateRequest, null, [{
    key: "initialize",
    value: function initialize(obj, userProviderId, userId, companyId, marketplaceClientId) {
      obj['user_provider_id'] = userProviderId;
      obj['user_id'] = userId;
      obj['company_id'] = companyId;
      obj['marketplace_client_id'] = marketplaceClientId;
    }

    /**
     * Constructs a <code>UserProviderLinkCreateRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserProviderLinkCreateRequest} obj Optional instance to populate.
     * @return {module:model/UserProviderLinkCreateRequest} The populated <code>UserProviderLinkCreateRequest</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserProviderLinkCreateRequest();
        if (data.hasOwnProperty('user_provider_id')) {
          obj['user_provider_id'] = _ApiClient["default"].convertToType(data['user_provider_id'], 'String');
          delete data['user_provider_id'];
        }
        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }
        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }
        if (data.hasOwnProperty('marketplace_client_id')) {
          obj['marketplace_client_id'] = _ApiClient["default"].convertToType(data['marketplace_client_id'], 'String');
          delete data['marketplace_client_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UserProviderLinkCreateRequest;
}();
/**
 * Unique identifier linking a user to the installed integration. Generated by the integration.
 * @member {String} user_provider_id
 */
UserProviderLinkCreateRequest.prototype['user_provider_id'] = undefined;

/**
 * Pipedrive user ID
 * @member {Number} user_id
 */
UserProviderLinkCreateRequest.prototype['user_id'] = undefined;

/**
 * Pipedrive company ID
 * @member {Number} company_id
 */
UserProviderLinkCreateRequest.prototype['company_id'] = undefined;

/**
 * Pipedrive Marketplace client ID of the installed integration
 * @member {String} marketplace_client_id
 */
UserProviderLinkCreateRequest.prototype['marketplace_client_id'] = undefined;
var _default = UserProviderLinkCreateRequest;
exports["default"] = _default;
