// pipedrive@23.4.3/dist/model/UserProviderLinkSuccessResponseData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UserProviderLinkSuccessResponseData model module.
 * @module model/UserProviderLinkSuccessResponseData
 * @version 1.0.0
 */
var UserProviderLinkSuccessResponseData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserProviderLinkSuccessResponseData</code>.
   * @alias module:model/UserProviderLinkSuccessResponseData
   */
  function UserProviderLinkSuccessResponseData() {
    (0, _classCallCheck2["default"])(this, UserProviderLinkSuccessResponseData);
    UserProviderLinkSuccessResponseData.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UserProviderLinkSuccessResponseData, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserProviderLinkSuccessResponseData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserProviderLinkSuccessResponseData} obj Optional instance to populate.
     * @return {module:model/UserProviderLinkSuccessResponseData} The populated <code>UserProviderLinkSuccessResponseData</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserProviderLinkSuccessResponseData();
        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
          delete data['message'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UserProviderLinkSuccessResponseData;
}();
/**
 * The success message of the request
 * @member {String} message
 */
UserProviderLinkSuccessResponseData.prototype['message'] = undefined;
var _default = UserProviderLinkSuccessResponseData;
exports["default"] = _default;
