// pipedrive@23.4.3/dist/model/UserProviderLinkErrorResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UserProviderLinkErrorResponse model module.
 * @module model/UserProviderLinkErrorResponse
 * @version 1.0.0
 */
var UserProviderLinkErrorResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserProviderLinkErrorResponse</code>.
   * @alias module:model/UserProviderLinkErrorResponse
   */
  function UserProviderLinkErrorResponse() {
    (0, _classCallCheck2["default"])(this, UserProviderLinkErrorResponse);
    UserProviderLinkErrorResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UserProviderLinkErrorResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserProviderLinkErrorResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserProviderLinkErrorResponse} obj Optional instance to populate.
     * @return {module:model/UserProviderLinkErrorResponse} The populated <code>UserProviderLinkErrorResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserProviderLinkErrorResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
          delete data['message'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UserProviderLinkErrorResponse;
}();
/**
 * Boolean that indicates whether the request was successful or not
 * @member {Boolean} success
 */
UserProviderLinkErrorResponse.prototype['success'] = undefined;

/**
 * The error message of the request
 * @member {String} message
 */
UserProviderLinkErrorResponse.prototype['message'] = undefined;
var _default = UserProviderLinkErrorResponse;
exports["default"] = _default;
